src/storage/txn/latch.rs

```diff
@@ -75,6 +73,31 @@ impl Latch {
         None
     }
 
+    /// Clear the waiting info with specific `key_hash` and `who`.
+    /// If the element which would be removed does not appear at the
+    /// front of the queue, it will leave a hole in the queue. So we must remove
+    /// consecutive hole when remove the head of the queue to make the queue not
+    /// too long.
+    pub fn clear_waiting(&mut self, key_hash: u64, who: u64) -> Option<(u64, u64)> {
+        if let Some(item) = self.waiting.pop_front() {
+            if let Some((k, cid)) = item.as_ref() {
+                if *k == key_hash && *cid == who {
+                    self.maybe_shrink();
+                    return item;
+                }
+                self.waiting.push_front(item);
+            }
+            for it in self.waiting.iter_mut() {
+                if let Some((v, cid)) = it {
+                    if *v == key_hash && *cid == who {
+                        return it.take();
```

Comment:
Can we use `VecDequeue::remove` so that it does not leave holes? Then we don't have to take the head of the queue as a special case.